base/daemon/SentryWriter.cpp

```diff
@@ -0,0 +1,244 @@
+        sentry_options_set_database_path(options, temp_folder_path.c_str());
+        if (strstr(VERSION_DESCRIBE, "-stable") || strstr(VERSION_DESCRIBE, "-lts"))
+        {
+            sentry_options_set_environment(options, "prod");
+        }
+        else
+        {
+            sentry_options_set_environment(options, "test");
+        }
+
+        const std::string & http_proxy = config.getString("send_crash_reports.http_proxy", "");
+        if (!http_proxy.empty())
+        {
+            sentry_options_set_http_proxy(options, http_proxy.c_str());
+        }
+
+        int init_status = sentry_init(options);
+        if (!init_status)
+        {
+            initialized = true;
+            anonymize = config.getBool("send_crash_reports.anonymize", false);
+            const std::string& anonymize_status = anonymize ? " (anonymized)" : "";
+            LOG_INFO(
+                logger,
+                "Sending crash reports is initialized with {} endpoint and {} temp folder{}",
+                endpoint,
+                temp_folder_path,
+                anonymize_status);
+        }
+        else
+        {
+            LOG_WARNING(logger, "Sending crash reports failed to initialized with {} status", init_status);
+        }
+    }
+    else
+    {
+        LOG_INFO(logger, "Sending crash reports is disabled");
+    }
+#else
+    UNUSED(config);
+#endif
+}
+
+void SentryWriter::shutdown()
+{
+#if USE_SENTRY
+    if (initialized)
+    {
+        sentry_shutdown();
+    }
+#endif
+}
+
+void SentryWriter::onFault(int sig, const siginfo_t & info, const ucontext_t & context, const StackTrace & stack_trace)
+{
+#if USE_SENTRY
+    auto * logger = &Poco::Logger::get("SentryWriter");
+    if (initialized)
+    {
+        const std::string & error_message = signalToErrorMessage(sig, info, context);
+        sentry_value_t event = sentry_value_new_message_event(SENTRY_LEVEL_FATAL, "fault", error_message.c_str());
+        sentry_set_tag("signal", strsignal(sig));
+        sentry_set_extra("signal_number", sentry_value_new_int32(sig));
+        setExtras();
+
+        /// Prepare data for https://develop.sentry.dev/sdk/event-payloads/stacktrace/
+        sentry_value_t sentry_frames = sentry_value_new_list();
+        size_t stack_size = stack_trace.getSize();
+        if (stack_size > 0)
+        {
+            size_t offset = stack_trace.getOffset();
+            if (stack_size == 1)
+            {
+                offset = 1;
+            }
+            char instruction_addr[100];
+            StackTrace::Frames frames;
+            StackTrace::symbolize(stack_trace.getFramePointers().data(), offset, stack_size, frames);
+            for (size_t i = stack_size - 1; i >= offset; --i)
+            {
+                const StackTrace::Frame & current_frame = frames[i];
+                sentry_value_t sentry_frame = sentry_value_new_object();
+                UInt64 frame_ptr = reinterpret_cast<UInt64>(current_frame.virtual_addr);
```

Comment:
Maybe we can format pointer without a cast to UInt64?

Reply:
> Maybe we can format pointer without a cast to UInt64?

@alexey-milovidov maybe, but it just mimics what they do in alternative ways to collect stacktraces (which are not currently used).